### PR TITLE
KS with preferred host support

### PR DESCRIPTION
**Pull request by p0lyn0mial**
to switch KS to use a preferred host (for now only `localhost` is supported) the following `unsupported-kube-api-over-localhost` must be set to `true`

```
unsupportedConfigOverrides:
  arguments:
    unsupported-kube-api-over-localhost:
      - "true"
```

The preferred host is the primary host to which KS will be sending all requests. There is a health monitor that is constantly monitoring it. When the new host is unavailable all requests will be directed to the old host (from kubeconfig)

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
requires https://github.com/openshift/library-go/pull/1030

File: cmd/kube-scheduler/app/config/patch.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import (
+	"k8s.io/client-go/transport"
+
+	"github.com/openshift/library-go/pkg/monitor/health"
+)
+
+// OpenShiftContext is additional context that we need to launch the kube-scheduler for openshift.
+// Basically, this holds our additional config information.
+type OpenShiftContext struct {
+	UnsupportedKubeAPIOverPreferredHost bool
```
Comment (mfojtik) [inline cmd/kube-scheduler/app/config/patch.go:12]: can we add one-line godoc above these?

**Comment by mfojtik** [comment]
one nit, LGTM otherwise

**Pull request merged**